llvm/lib/Analysis/CheriBounds.cpp

```diff
@@ -13,14 +13,18 @@ using namespace llvm;
   DBG_MESSAGE(llvm::right_justify("-", Depth + 1) << __VA_ARGS__)
 
 // This is similar to CaptureTracking but we treat way more cases as "captured"
-bool CheriNeedBoundsChecker::check(const Use &U) const {
+bool CheriNeedBoundsChecker::check(const Use &U, bool Simple) const {
   if (!MinSizeInBytes) {
     DBG_MESSAGE("dyanmic size alloca always needs bounds: ";
                 U.getUser()->dump());
     return true;
   }
   APInt CurrentGEPOffset = APInt(DL.getIndexSizeInBits(PointerAS), 0);
-  bool Result = useNeedsBounds(U, CurrentGEPOffset, 1);
```

Comment:
Could you add a test for this change? For example a function with three allocas, load, store, gep?

Reply:
As in an explicit optnone test that illustrates bounds being omitted for load+store but not GEP+load/store?